docs/source/redirects.py

```
@@ -99,23 +192,45 @@ def _process_static_redirects(app):
             )
 
 
+def _write_hash_redirect(app, base_page, anchor_map, default_url):
+    """Write a hash-based redirect page for `base_page`."""
+    anchor_map_json = json.dumps(anchor_map, indent=2)
+
+    logger.info(
+        "Creating hash redirect page '%s' with %d anchors"
```

Ruff (0.16.0)

[warning] Use format specifiers instead of percent format

(UP031)